src/regex.cr

```diff
@@ -246,14 +250,35 @@ class Regex
     # :nodoc:
     UCP = 0x2000_0000
 
+    # Force pattern anchoring at the end of the subject.
+    #
+    # Unsupported with PCRE.
     ENDANCHORED = 0x8000_0000
+  end
+
+  alias CompileOptions = Options
+
+  # Represents options passed to regex match methods such as `Regex#match`.
+  @[Flags]
+  enum MatchOptions
```

Comment:
Can a user manually pass `NO_UTF_CHECK` after #13240 is applied? If yes it needs a corresponding enum member here

Reply:
I don't want to add it here because it only makes sense with #13240. I'm planning to add it there.
Although it won't actually have any effect on PCRE2 >= 10.34 unless we provide an option to opt-out of `MATCH_INVALID_UTF` (https://github.com/crystal-lang/crystal/issues/13237#issuecomment-1489844998).